common/lib/util/eventemitter.js

```diff
@@ -57,7 +57,7 @@ var EventEmitter = (function() {
 			/* ... or we take event to be the actual event name and listener to be all */
 		}
 		var listeners, idx = -1;
-		if(event === null) {
+		if((event === null) || (event === undefined)) {
```

Comment:
OK, I partially take back what I said because now we have a new can of worms because this method, when called from `RealtimeChannel.unsubscribe()` can no longer rely on its own logic to know which overloaded signature applies, because that was done before it was called. So the overloading logic is re-performed here which is a bit crap; ideally we'd have an `_off()` or something which knows it doesn't need to do that. It's not such an issue for `on()`/`off()` though.

That aside, we now have one or more values for `event` which signify that no `event` argument was supplied, and the intended meaning is that the overloaded variant with no `event` argument - ie unsubscribe from all events - was called. The special value could be `null` or `undefined` but while this is only an argument passed to an internal method it is moot..

However, the question this raises is whether or not we intend that this behaviour also forms part of the public API: is it intended that a client can pass `null`, or `undefined`, or any falsy value, and thereby request to unsubscribe from all events?

The standard that defines best practice for these situations is WebIDL: http://www.w3.org/TR/WebIDL/

I think we should in fact expose an API publically which is in WebIDL terminology 

    off([TreatUndefinedAs=Null] DOMString? event, Listener? listener)

where the `event` argument, if `null` (IDL null, not JS `null`) means unsubscribe from all events.

Now WebIDL tells us which JS argument values map to IDL `null`: they are `undefined` and `null`.

The other falsy values map `false` -> `'false'`, `0` -> `'0'`, `''` -> `''`; ie they are naturally coerced to a string if not already a string.

The upshot is that we should be checking for `(x === null || x === undefined)` instead of `!x`, and it's fine (and arguably more WebIDL-esque) for the overload-resolution functionality in the derived class to pass `null`.